openedx/adg/admin_dashboard/performance/views.py

```diff
@@ -1,3 +1,149 @@
 """
 Views for `performance` app
 """
+from datetime import date
+
+from django.contrib.auth.mixins import UserPassesTestMixin
+from django.contrib.auth.models import User
+from django.db.models import Prefetch, Q
+from django.views.generic import ListView, View
+from opaque_keys.edx.keys import CourseKey
+
+from common.djangoapps.student.models import CourseEnrollment
+from lms.djangoapps.courseware.courses import get_course_by_id
+from openedx.adg.admin_dashboard.constants import DEFAULT_PAGINATE_BY
+from openedx.adg.admin_dashboard.helpers import get_http_response_with_csv_file_attachment
+from openedx.adg.admin_dashboard.performance.helpers import get_course_grades_for_users, get_course_grades_report_data
+from openedx.adg.lms.applications.models import BusinessLine, MultilingualCourseGroup
+from openedx.adg.lms.applications.rules import is_adg_admin
+
+
+class CourseGradesView(UserPassesTestMixin, ListView, View):
+    """
+    View to list and download grades of a course.
+    """
+
+    course = None
+    template_name = 'adg/admin_dashboard/performance/course_grades.html'
+    http_method_names = ('get', 'post')
+
+    def test_func(self):
+        """
+        Checks user permissions to access course grades.
+
+        Authenticated and staff user will be able to access course grades if he/she is a
+        superuser | adg admin | bu admin and course is a bu prerequisite for the same bu.
+
+        Returns:
+            bool: True if user has permission, False otherwise
+        """
+        user = self.request.user
+        if not (user.is_authenticated and user.is_staff):
+            return False
+
+        if user.is_superuser or is_adg_admin(user):
+            return True
+
+        if not user.groups.filter(id__in=BusinessLine.objects.values_list('group__id')).exists():
+            return False
+
+        course_key = CourseKey.from_string(self.kwargs['course_key_string'])
```

Comment:
The same course_key is being extracted in line 147-148 under the `_get_course` method.

You can introduce a new function like `_get_course_key` and put this logic in there and call that function here as well as in the `setup` method or in the `_get_course` method.

Reply:
Something like:

```
def _get_course_key(self):
    return CourseKey.from_string(self.kwargs['course_key_string'])
```